api/ccapi/include/layer.h

```diff
@@ -63,159 +63,56 @@ class Layer {
-
-  /**
-   * @brief     check hyper parameter for the layer
-   * @retval #ML_ERROR_NONE Successful.
-   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
-   */
-  virtual int checkValidation() = 0;
-
-  /**
-   * @brief  get the loss value added by this layer
-   * @retval loss value
-   */
-  virtual float getLoss() = 0;
-
-  /**
-   * @brief     set trainable for this layer
-   * @param[in] train to enable/disable train
-   */
-  virtual void setTrainable(bool train) = 0;
-
-  /**
-   * @brief     get if the output of this layer must be flatten
-   * @retval    flatten value
-   */
-  virtual bool getFlatten() = 0;
-
-  /**
-   * @brief     Get name of the layer
-   */
-  virtual std::string getName() noexcept = 0;
-
   /**
    * @brief Get the layer type
    * @return const std::string type representation
    */
   virtual const std::string getType() const = 0;
 
   /**
-   * @brief Preset modes for printing summary for the layer
+   * @brief     Default allowed properties
+   * - input shape : string
+   * - bias zero : bool
+   * - normalization : bool
+   * - standardization : bool
+   * - activation : string (type)
+   * - epsilon : float
+   * - weight_regularizer : string (type)
+   * - weight_regularizer_constant : float
+   * - unit : int
+   * - weight_initializer : string (type)
+   * - filter_size : int
+   * - kernel_size : ( n , m )
+   * - stride : ( n, m )
+   * - padding : ( n, m )
+   * - pool_size : ( n,m )
+   * - pooling : max, average, global_max, global_average
+   * - flatten : bool
+   * - name : string (type)
+   * - num_inputs : unsigned int (minimum 1)
+   * - num_outputs : unsigned int (minimum 1)
+   * - momentum : float,
+   * - moving_mean_initializer : string (type),
+   * - moving_variance_initializer : string (type),
+   * - gamma_initializer : string (type),
+   * - beta_initializer" : string (type)
+   * - modelfile : model file for loading config for backbone layer
+   * - input_layers : string (type)
+   * - output_layers : string (type)
+   * - trainable :
+   * - flip_direction
+   * - random_translate
+   * - in_dim : int ( input dimension for embedding layer )
+   * - out_dim : int ( output dimesion for embedding layer )
+   * - in_length : int ( input length for embedding layer )
    */
-  enum class PrintPreset {
-    PRINT_NONE = 0,     /**< Print nothing */
-    PRINT_SUMMARY,      /**< Print preset including summary information */
-    PRINT_SUMMARY_META, /**< Print summary preset that includes meta information
-                         */
-    PRINT_ALL           /**< Print everything possible */
-  };
-
   /**
-   * @brief print using PrintPreset
-   *
-   * @param out oustream
-   * @param preset preset to be used
+   * @brief     set Property of layer
+   * @param[in] values values of property
+   * @retval #ML_ERROR_NONE Successful.
+   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
    */
-  virtual void printPreset(std::ostream &out,
```

Comment:
shouldn't print remain here?

Reply:
The ccAPI for now must be equivalent to cAPI.

Reply:
Acked